source/includes/inkling2-reference/_simulators.html.md

```diff
@@ -0,0 +1,89 @@
+# Simulators
+
+```inkling2--code
+simulator MySimulator(action: MyActionType, config: MyConfigType): MyStateType {
+}
+```
+
+Curriculum statements reference an associated simulator in the [source clause][3].
+
+The `simulator` statement describes the interface to a simulator program. The Inkling program does not contain code for the simulator itself. Instead the Inkling program defines how the simulator is used to train Inkling concepts.
+
+> Simulator Declaration Syntax
+
+```inkling2--syntax
+simulator <simulatorName>'('action <typeReference>
+  [',' config <ctypeReference>]? ')' ':' <typeReference> '{'
+'}'
+```
+
+A `simulator` statement defines the interface to the simulator - its input and output parameter types.
+
+* The action input type
+* An optional configuration input type
```

Comment:
Note that optional config seems to be broken (#9944). Maybe we can still leave it here and just fix the bug.